### PR TITLE
Feature WPML support

### PR DESCRIPTION
**Pull request by pereirinha**
Add support for WPML

### CODE REVIEW

File: php/class-assets.php [hunk 1]
```diff
@@ -1020,7 +1020,8 @@ public function get_asset( $post_id, $type = 'object' ) {
 		global $wpdb;
 
 		$wpdb->cld_table = Utils::get_relationship_table();
-		$prepare         = $wpdb->prepare( "SELECT * FROM $wpdb->cld_table WHERE post_id = %d;", (int) $post_id );
+		$media_context   = Utils::get_media_context( $post_id );
+		$prepare         = $wpdb->prepare( "SELECT * FROM $wpdb->cld_table WHERE post_id = %d AND media_context = %s;", (int) $post_id, $media_context );
```
Comment (yuval-cloudinary) [inline php/class-assets.php:1024]: Should we add media_context to the query in https://github.com/cloudinary/cloudinary_wordpress/blob/d2bfec53c3144f02f20e80fa75c118db5c635786/php/sync/class-sync-queue.php#L325 ?

File: php/relate/class-relationship.php [hunk 2]
```diff
@@ -60,17 +69,20 @@ public function __construct( $post_id ) {
 	 * @return array|null
 	 */
 	public function get_data() {
-		$cache_data = wp_cache_get( $this->post_id, Sync::META_KEYS['cloudinary'] );
+		$key = $this->post_id . $this->context;
```
Comment (yuval-cloudinary) [inline php/relate/class-relationship.php:72]: I would consider adding a separator like a dash or something
maybe adding sanitize_key() as well

**Review by yuval-cloudinary — CHANGES_REQUESTED** [review]

File: php/class-utils.php [hunk 3]
```diff
@@ -417,6 +428,24 @@ public static function upgrade_3_0_1() {
 		update_option( Sync::META_KEYS['db_version'], get_plugin_instance()->version );
 	}
 
+	/**
+	 * Upgrade DB from v3.0.1 to v3.1.9.
+	 */
+	public static function upgrade_3_1_9() {
```
Comment (yuval-cloudinary) [inline php/class-utils.php:434]: It is interesting that the install path is a straightforward dbDelta, and the upgrade path goes through custom logic.
I would expect to go through exactly the same db scheme upgrade regardless if I uninstalled or not before upgrading

File: php/class-assets.php (hunk 1 above)
Comment (pereirinha) [reply php/class-assets.php:1024]: That's a good question @yuval-cloudinary.

The single purpose of that query you're pointing out is to get the count of unsynced assets to populate the plugin dashboard. The way it is, it's still retrieving all the assets, regardless of the context. If we filter by context, we might get a partial overlook that doesn't reflect the overall state.

File: php/relate/class-relationship.php (hunk 2 above)
Comment (pereirinha) [reply php/relate/class-relationship.php:72]: I can do that, but this key is not public facing, so the benefits are questionable. What I'll do, is to ensure that the return for `get_media_context()` is sanitized.

File: php/class-utils.php (hunk 3 above)
Comment (pereirinha) [reply php/class-utils.php:434]: Hum, not sure if I follow.

Comment (yuval-cloudinary) [reply php/class-utils.php:434]: sorry, after rereading the code this is a non-issue

**Review by yuval-cloudinary — APPROVED** [review]

**Pull request merged**